smoke.py

```diff
@@ -193,5 +202,91 @@ def testUserManipulation(self):
             self._check_cmd_result(cmd, test_ret, test_output, expect_pass=False)
 
 
+    def _prepare_compiler_test_directory(self):
+
+        # create a temporary directory
+        tmpdir = tempfile.mkdtemp()
+
+        self.log.info("Compiler test temporary directory is %s" % tmpdir)
+
+        # Copy the `hello.sh` script from this resource directory into the
+        # temporary directory
+        src = os.path.join(self.compiler_resource_dir, "hello.sh")
+        dest = os.path.join(tmpdir, "hello.sh")
+        try:
+            shutil.copy(src, dest)
+        except shutil.Error as e:
+            self.log.info('Error: %s' % e)
+        except IOError as e:
+            self.log.info('Error: %s' % e.strerror)
+
+        # make sure destination script is executable
+        st = os.stat(dest)
+        os.chmod(dest, st.st_mode | stat.S_IEXEC)
+
+        # Place a gzipped tarball of `hello.c` and `Makefile` from the
+        # resource directory into the temporary directory with the name
+        # `hello.tgz`.
+        dest = os.path.join(tmpdir, "hello.tgz")
+        tar = tarfile.open(dest, "w:gz")
+        for f in ["hello.c", "Makefile"]:
+            src = os.path.join(self.compiler_resource_dir, f)
+            tar.add(src, arcname=f)
+        tar.close()
+
+        self.compiler_test_dir = tmpdir
+
+    def _cleanup_compiler_test_directory(self):
+
+        # clean up the temporary directory
+        if self.compiler_test_dir:
+            self.log.info("cleaning up compiler test directory")
+            shutil.rmtree(self.compiler_test_dir)
```

Comment:
You need to handle the case where `shutil.rmtree(self.compiler_test_dir)` fails. For example we might get a permission error or the directory doesn't exist yet when tearDown is called.

That said, the risk of permission errors and similar happening are very small, so for the purposes of this test, it should probably be okay to just do

`shutil.rmtree(self.compiler_test_dir, ignore_errors=True)`